src/main.original.js

```diff
@@ -1,204 +1,288 @@
-  hTxt.textContent='장바구니';
-  addBtn.textContent='추가';
-  updateSelOpts();
+  let cont = document.createElement('div');
+  cont.className = 'bg-gray-100 p-8';
+
+  let wrap = document.createElement('div');
+  wrap.className = 'max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl p-8';
+
+  let hTxt = document.createElement('h1');
+  hTxt.className = 'text-2xl font-bold mb-4';
+  hTxt.textContent = '장바구니';
+
+  cartItem_Elmt = document.createElement('div');
+  cartItem_Elmt.id = 'cart-items';
+
+  totalPrice_Elmt = document.createElement('div');
+  totalPrice_Elmt.id = 'cart-total';
+  totalPrice_Elmt.className = 'text-xl font-bold my-4';
+
+  itemDropbox_Elmt = document.createElement('select');
+  itemDropbox_Elmt.id = 'product-select';
+  itemDropbox_Elmt.className = 'border rounded p-2 mr-2';
+
+  addBtn_Elmt = document.createElement('button');
+  addBtn_Elmt.id = 'add-to-cart';
+  addBtn_Elmt.className = 'bg-blue-500 text-white px-4 py-2 rounded';
+  addBtn_Elmt.textContent = '추가';
+
+  stock_Elmt = document.createElement('div');
+  stock_Elmt.id = 'stock-status';
+  stock_Elmt.className = 'text-sm text-gray-500 mt-2';
+
+  setSaleItemDropbox();
   wrap.appendChild(hTxt);
-  wrap.appendChild(cartDisp);
-  wrap.appendChild(sum);
-  wrap.appendChild(sel);
-  wrap.appendChild(addBtn);
-  wrap.appendChild(stockInfo);
+  wrap.appendChild(cartItem_Elmt);
+  wrap.appendChild(totalPrice_Elmt);
+  wrap.appendChild(itemDropbox_Elmt);
+  wrap.appendChild(addBtn_Elmt);
+  wrap.appendChild(stock_Elmt);
   cont.appendChild(wrap);
-  root.appendChild(cont);
-  calcCart();
+  document.getElementById('app').appendChild(cont);
+
+  calcTotalPrice();
+
+  // 번개세일 팝업
   setTimeout(function () {
     setInterval(function () {
-      var luckyItem=prodList[Math.floor(Math.random() * prodList.length)];
-      if(Math.random() < 0.3 && luckyItem.q > 0) {
-        luckyItem.val=Math.round(luckyItem.val * 0.8);
+      let luckyItem = saleItemList[Math.floor(Math.random() * saleItemList.length)];
+
+      if (Math.random() < 0.3 && luckyItem.qty > 0) {
+        let 할인20 = 0.8;
+        luckyItem.price = Math.round(luckyItem.price * 할인20);
         alert('번개세일! ' + luckyItem.name + '이(가) 20% 할인 중입니다!');
-        updateSelOpts();
+        setSaleItemDropbox();
       }
-    }, 30000);
+    }, 30000); //30s
   }, Math.random() * 10000);
+
+  // 구매 제안 팝업
   setTimeout(function () {
     setInterval(function () {
-      if(lastSel) {
-        var suggest=prodList.find(function (item) { return item.id !== lastSel && item.q > 0; });
-        if(suggest) {
-          alert(suggest.name + '은(는) 어떠세요? 지금 구매하시면 5% 추가 할인!');
-          suggest.val=Math.round(suggest.val * 0.95);
-          updateSelOpts();
-        }
+      if (!lastAddedItem) return;
+
+      //추천상품 : 최근 구매한거 아니고, 양 0보다 많은거
+      let suggest = saleItemList.find(function (item) {
+        return item.id !== lastAddedItem && item.qty > 0;
+      });
```

Comment:
let -> const 랑 Arrow Function 을 쓰면 가독성이 더 좋을 것 같습니다
보통 인자값은 당연히 배열의 아이템이라 저는 종종 짧게 써요
예를 들면 아래처럼 씁니다유

```suggestion
      const suggest = saleItemList.find((x) => x.id !== lastAddedItem && x.qty > 0);
```